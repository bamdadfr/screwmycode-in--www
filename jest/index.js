export { JestWithStyledComponents } from './jest-with-styled-components'