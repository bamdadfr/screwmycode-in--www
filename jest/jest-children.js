import React from 'react'

/**
 * @function
 * @name JestChildren
 * @description jest: dummy children
 * @returns {React.ReactElement} - react component
 */
export function JestChildren () {

    return (
        <>
            <span>children</span>
            <span>children</span>
            <span>children</span>
        </>
    )

}