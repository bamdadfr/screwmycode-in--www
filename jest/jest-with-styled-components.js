import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Global, Theme } from '../src/app/styles'

/**
 * @function
 * @name JestWithStyledComponents
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactElement} Component - react sub component
 * @returns {React.ReactElement} - react component
 */
export function JestWithStyledComponents (Component) {

    return (
        <>
            <Global/>
            <ThemeProvider theme={Theme}>
                {Component}
            </ThemeProvider>
        </>
    )

}