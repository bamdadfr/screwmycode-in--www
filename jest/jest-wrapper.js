import React from 'react'
import { RecoilRoot } from 'recoil'
import { ThemeProvider } from 'styled-components'
import { ThemeStyles } from '../src/styles/theme.styles'

/**
 * @function
 * @name JestWrapper
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactNode} Component - react sub component
 * @returns {React.ReactNode} - react component
 */
export function JestWrapper (Component) {

    return (
        <RecoilRoot>
            <ThemeProvider theme={ThemeStyles}>
                {Component}
            </ThemeProvider>
        </RecoilRoot>
    )

}