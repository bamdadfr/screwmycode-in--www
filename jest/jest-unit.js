import React from 'react'
import { shallow } from 'enzyme'
import { JestWithStyledComponents } from './jest-with-styled-components'

/**
 * @function
 * @name JestUnit
 * @description jest: get unit from enzyme
 * @param {React.ReactElement} Component - react component
 * @param {boolean} [wrap=false] - wrap with state and theme...
 * @returns {object} - enzyme shallow object
 */
export function JestUnit (Component, wrap = false) {

    if (wrap) {

        return shallow (
            JestWithStyledComponents (Component),
        )

    }

    return shallow (Component)

}