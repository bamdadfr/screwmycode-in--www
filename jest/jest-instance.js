import renderer from 'react-test-renderer'
import { JestWrapper } from './jest-wrapper'

export const JestInstance = (Component) => renderer.create (JestWrapper (Component)).toJSON ()