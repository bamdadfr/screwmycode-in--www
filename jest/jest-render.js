import React from 'react'
import { render } from '@testing-library/react'
import { JestWrapper } from './jest-wrapper'

/**
 * @function
 * @name JestRender
 * @description jest: get a render from testing-library
 * @param {React.ReactElement} Component - react component to render
 * @returns {object} - testing library render
 */
export function JestRender (Component) {

    return render (
        JestWrapper (Component),
    )

}