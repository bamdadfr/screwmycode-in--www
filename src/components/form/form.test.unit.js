import React from 'react'
import FormComponent from './form.component'
import { JestInstance } from '../../../jest/jest-instance'

const instance = JestInstance (<FormComponent/>)

it ('should mount and match', () => {

    expect (instance).toBeDefined ()

    expect (instance).toMatchSnapshot ()

})

it ('should have a classname prop from styled components', () => {

    expect (instance.props.className).toBeDefined ()

    expect (instance.props.className).toContain ('sc')

})

it ('should have 2 children', () => {

    expect (instance.children).toHaveLength (2)

})