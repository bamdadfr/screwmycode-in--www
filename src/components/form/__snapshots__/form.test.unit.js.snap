// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount and match 1`] = `
<form
  className="sc-1forcsr-0 fRPgeZ"
  onSubmit={[Function]}
>
  <input
    className="sc-1forcsr-1 iBFTf"
    onChange={[Function]}
    placeholder="insert youtube link here"
    type="text"
    value=""
  />
  <input
    className="sc-1forcsr-2 fyEvAn"
    type="submit"
    value="submit"
  />
</form>
`;
